mne_sandbox/preprocessing/_sns.py

```diff
@@ -0,0 +1,175 @@
+# -*- coding: utf-8 -*-
+"""Sensor noise suppression"""
+
+import numpy as np
+
+from mne import compute_raw_covariance
+from mne.io.pick import _pick_data_channels, pick_channels
+from mne.io import _BaseRaw
+from mne.utils import logger, verbose
+
+from ._dss import _pca
+
+
+class SensorNoiseSuppression(object):
+    """Apply the sensor noise suppression (SNS) algorithm
+
+    This algorithm (from [1]_) will replace the data from each channel by
+    its regression on the subspace formed by the other channels.
+
+    .. note:: Bad channels are not modified or reset by this function.
+
+    Parameters
+    ----------
+    n_neighbors : int
```

Comment:
I think in the matlab code, there are two input params `n_neighbors` and `n_skip`. Sometimes you don't want to use correlation with the immediate neighbors because neighboring sensors can get corrupted together (I think that's the reason for it ...)